src/pages/auth/SignupPage.tsx

```diff
@@ -43,46 +34,59 @@ export const SignupPage: React.FC = () => {
         <Typography component="h1" variant="h5">
           Sign up
         </Typography>
-        {error && (
-          <Alert severity="error" sx={{ mt: 2, width: "100%" }}>
-            {error}
-          </Alert>
-        )}
-        <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 3 }}>
+        <Box component="form" onSubmit={handleSubmit(onSubmit)} noValidate sx={{ mt: 3 }}>
           <TextField
             margin="normal"
             required
             fullWidth
-            id="name"
             label="Full Name"
-            name="name"
             autoComplete="name"
             autoFocus
-            value={name}
-            onChange={(e) => setName(e.target.value)}
+            {...register("name", {
+              required: "Name is a required field!",
+            })}
+            error={!!errors.name}
+            helperText={errors.name?.message}
           />
           <TextField
             margin="normal"
             required
             fullWidth
             id="email"
+            type="email"
             label="Email Address"
-            name="email"
             autoComplete="email"
-            value={email}
-            onChange={(e) => setEmail(e.target.value)}
+            {...register("email", {
+              required: "Email is a required field!", pattern: {
+                value: /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/,
+                message: "Please enter a valid email address!",
+              },
+            },)}
+            error={!!errors.email}
+            helperText={errors.email?.message}
           />
           <TextField
             margin="normal"
             required
             fullWidth
-            name="password"
             label="Password"
             type="password"
             id="password"
             autoComplete="new-password"
-            value={password}
-            onChange={(e) => setPassword(e.target.value)}
+            {
+            ...register("password", {
+              required: "Password is a required field!", minLength: {
```

Comment:
Use zod for validation instead of inline validation

Reply:
Done!